### ISSUE TITLE
感谢分享，我想请教一个问题，请问人脸检测精选的mxnet模型如何成功的转换成ncnn的模型的啊？

### ISSUE THREAD

**Issue by qaz734913414** [zuoqing1988/ZQCNN#64]
感谢分享，我想请教一个问题，请问人脸检测精选的mxnet模型如何成功的转换成ncnn的模型的啊？

**Comment by zuoqing1988** [comment]
手工删掉最后无用的几层，然后再根据实际情况确定param最前面的layer数量和blob数量

**Comment by qaz734913414** [comment]
> 7767517
17 18
Input            data                             0 1 data 0=16 1=16 2=3
Convolution      conv1                            1 1 data conv1 0=16 1=3 11=3 3=2 13=2 5=1 6=432
PReLU            prelu1                           1 1 conv1 prelu1 0=16
ConvolutionDepthWise conv2_dw                         1 1 prelu1 conv2_dw 0=16 1=3 11=3 3=2 13=2 5=1 6=144 7=16
PReLU            prelu2_dw                        1 1 conv2_dw prelu2_dw 0=16
Convolution      conv2_sep                        1 1 prelu2_dw conv2_sep 0=24 1=1 11=1 5=1 6=384
PReLU            prelu2                           1 1 conv2_sep prelu2 0=24
ConvolutionDepthWise conv3_dw                         1 1 prelu2 conv3_dw 0=24 1=3 11=3 5=1 6=216 7=24
PReLU            prelu3_dw                        1 1 conv3_dw prelu3_dw 0=24
Split            splitncnn_0                      1 2 prelu3_dw prelu3_dw_splitncnn_0 prelu3_dw_splitncnn_1
Convolution      conv4_1                          1 1 prelu3_dw_splitncnn_1 conv4_1 0=2 1=1 11=1 5=1 6=48
BatchNorm        bn4_1                            1 1 conv4_1 bn4_1 0=2
Reshape          conv4_1_reshape                  1 1 bn4_1 conv4_1_reshape 0=2
Softmax          cls_prob                         1 1 conv4_1_reshape cls_prob 0=0
Convolution      conv4_2                          1 1 prelu3_dw_splitncnn_0 conv4_2 0=4 1=1 11=1 5=1 6=96
BatchNorm        bn4_2                            1 1 conv4_2 bn4_2 0=4
Reshape          conv4_2_reshape                  1 1 bn4_2 conv4_2_reshape 0=4

可以取出conv4_2的数据，但是cls_prob取不出数据，是不是不是这样改的啊？

**Comment by zuoqing1988** [comment]
可以先尝试我改好的，再对比对比就都会了
https://github.com/zuoqing1988/ZQCNN-MTCNN-vs-libfacedetection/tree/master/NCNN/model

**Comment by qaz734913414** [comment]
谢谢指导，我去试一下

**Comment by qaz734913414** [comment]
> convert cost: 2.972 ms
nms cost: 0.043 ms, (4-->0)
nms cost: 0.063 ms, (2-->0)
nms cost: 0.030 ms, (11-->1)
nms cost: 0.015 ms, (7-->0)
nms cost: 0.019 ms, (18-->2)
nms cost: 0.013 ms, (24-->4)
nms cost: 0.017 ms, (31-->5)
nms cost: 0.008 ms, (14-->3)
nms cost: 0.004 ms, (6-->1)
nms cost: 0.000 ms, (0-->0)
nms cost: 0.001 ms, (0-->0)
nms cost: 0.006 ms
first stage candidate count: 16
stage 1: cost 75.057 ms
run Rnet [7] times, candidate after nms: 1
stage 2: cost 7.097 ms
run Onet [1] times, candidate after nms: 1
stage 3: cost 7.073 ms
final found num: 1
total cost: 92.706 ms (P: 78.079 ms, R: 7.322 ms, O: 7.304 ms)
total 12.409 s / 100 = 124.089 ms

我尝试将https://github.com/zuoqing1988/ZQCNN-MTCNN-vs-libfacedetection
项目中NCNN跑了一下,编译成32位开启了openMp,没有开指令优化，速度只有124ms,设备配置是Intel(R) Pentium(R) CPU G2020 @ 2.90GHz
![image](https://user-images.githubusercontent.com/9314219/56180120-b1457680-603a-11e9-8525-d9a2c53f2867.png)
请问还可以优化速度不？

**Comment by zuoqing1988** [comment]
ncnn本身就没有x86的优化，你得用arm跑

**Comment by qaz734913414** [comment]
原来如此，看来需要移植到移动端再来看效果了，谢谢您的解答。
